src/pysdmx/model/__base.py

```diff
@@ -0,0 +1,212 @@
+    to derive classes because it is a subclass of AnnotableArtefact.
+
+    Attributes:
+        id: The identifier of the artefact.
+        uri: The URI of the artefact.
+        urn: The URN of the artefact.
+    """
+
+    id: str = ""
+    uri: Optional[str] = None
+    urn: Optional[str] = None
+
+
+class NameableArtefact(IdentifiableArtefact, frozen=True, omit_defaults=True):
+    """Nameable Artefact class.
+
+    Provides a name and a description to all derived classes.
+
+    Attributes:
+        name: The name of the artefact.
+        description: The description of the artefact.
+    """
+
+    name: Optional[str] = None
+    description: Optional[str] = None
+
+
+class VersionableArtefact(NameableArtefact, frozen=True, omit_defaults=True):
+    """Versionable Artefact class.
+
+    Provides a version to all derived classes.
+
+    Attributes:
+        version: The version of the artefact.
+        valid_from: The date from which the artefact is valid.
+        valid_to: The date to which the artefact is valid.
+    """
+
+    version: str = "1.0"
+    valid_from: Optional[datetime] = None
+    valid_to: Optional[datetime] = None
+
+
+class MaintainableArtefact(
+    VersionableArtefact, frozen=True, omit_defaults=True
+):
+    """Maintainable Artefact class.
+
+    An abstract class to group together primary structural metadata
+    artefacts that are maintained by an Agency.
+
+    Attributes:
+        is_final: Whether the artefact is final.
+        is_external_reference: Whether the artefact is an external reference.
+        service_url: The URL of the service.
+        structure_url: The URL of the structure.
+        agency: The maintainer of the artefact.
+    """
+
+    is_final: bool = False
+    is_external_reference: bool = False
+    service_url: Optional[str] = None
+    structure_url: Optional[str] = None
+    agency: Optional[Union[str, "Agency"]] = None
+
+
+class Item(NameableArtefact, frozen=True, omit_defaults=True):
+    """Item class.
+
+    The Item is an item of content in an Item Scheme. This may be a
+    concept in a concept scheme, a code in a codelist, etc.
+
+    Parent and child attributes (hierarchy) have been removed for simplicity.
+    """
+
+
+class ItemScheme(MaintainableArtefact, frozen=True, omit_defaults=True):
+    """ItemScheme class.
+
+    The descriptive information for an arrangement or division of objects
+    into groups based on characteristics, which the objects have in common.
+
+    Attributes:
+        items: The list of items in the scheme.
+        is_partial: Whether the scheme is partial.
+    """
+
+    items: Sequence["Item"] = ()
+    is_partial: bool = False
+
+
+class Organisation(Item, frozen=True, omit_defaults=True):
+    """Organisation class.
+
+    Attributes:
+        contacts: The contact of the agency.
+    """
+
+    contacts: Sequence["Contact"] = ()
+
+
+class Agency(Organisation, frozen=True, omit_defaults=True):
```

Comment:
We could also add DataProvider and DataConsumer for good measure. Happy to do it, once this PR has been merged :)

Reply:
Yes, please do it :)

Reply:
Mentioned in #33